llvm/include/llvm/IR/ConstantRangeList.h

```diff
@@ -72,6 +72,12 @@ class [[nodiscard]] ConstantRangeList {
                          APInt(64, Upper, /*isSigned=*/true)));
   }
 
+  void subtract(const ConstantRange &SubRange);
+  void subtract(int64_t Lower, int64_t Upper) {
```

Comment:
Is this mostly to simplify the test (line 121 in the test.cpp?)?

It might be better to put this logic in the test file instead (could be inlined into the line 121 area?), so that there is (a) smaller public API (b) less int64_t specific APIs (while the rest seem to be more generic regarding bitwidth)?

Reply:
Make sense. Deleted the API.